### PR TITLE
fix(TDI-44935): reuse custom query expression

### PR DESCRIPTION
**Pull request by ozhelezniak-talend**
* make query expression idempotent in the generated code
* use instead of duplication of custom query expression a variable
* store the expression in the variable and then read it again
* make obsolete the 'query' method with one argument

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-44935

**What is the new behavior?**
invoke query expression only once that prevents the change of logic for non-idempotent queries.

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard

**What kind of change does this PR introduce?**

- [x] Bugfix

**Does this PR introduce a breaking change?**

- [x] May. It depends on the customer's usage of the logger.

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet [hunk 1]
```diff
@@ -253,17 +253,18 @@ imports="
...
 		}
 		%>
 		java.sql.Statement stmt_<%=cid %> = conn_<%=cid %>.createStatement();
 		<%
 	}
-%>
-	String query_<%=cid %> = "";
-	boolean whetherReject_<%=cid%> = false;
\ No newline at end of file
+%>
```
Comment (Voievidko) [inline main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: I think it's not TDI component. Can we change it?

File: main/plugins/org.talend.designer.components.localprovider/components/templates/Log4j/Log4jDBConnUtil.javajet [hunk 2]
```diff
@@ -172,6 +172,16 @@ imports="
 			<%
 		}
 
+		public void query(INode node, String dbQueryVariableName){
+			beforeComponentProcess(node);
```
Comment (Voievidko) [inline main/plugins/org.talend.designer.components.localprovider/components/templates/Log4j/Log4jDBConnUtil.javajet:176]: In your new method you've skipped method with some action on query before login:
`org.talend.core.model.utils.NodeUtil.replaceCRLFInMEMO_SQL(dbquery);`
Will it have side effect for all components which use this util?

File: main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet (hunk 1 above)
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: I can revert the changes and remain non-idempotent logic for tHiveRow component.

Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/tHiveRow/tHiveRow_begin.javajet]: Let's revert this file. We also can create a work item for BD team to point them to this problem.

File: main/plugins/org.talend.designer.components.localprovider/components/templates/Log4j/Log4jDBConnUtil.javajet (hunk 2 above)
Comment (ozhelezniak-talend) [reply main/plugins/org.talend.designer.components.localprovider/components/templates/Log4j/Log4jDBConnUtil.javajet:176]: The invoker of the method should take care of it. "dbquery" variable should be already honed with it.
For example, it already honed by dbRowBeginUtil.getQueryString(node)
tAbstractDBRow_begin.javajet -> HelpClass.javajet
https://github.com/Talend/tdi-studio-se/blob/f749ac3e2a8627fc478f2f926364e809674dd1ab/main/plugins/org.talend.designer.components.localprovider/components/templates/DB/Row/HelpClass.javajet#L135

Comment (Voievidko) [reply main/plugins/org.talend.designer.components.localprovider/components/templates/Log4j/Log4jDBConnUtil.javajet:176]: Clear!

**Review by Voievidko — APPROVED** [review]
LGTM

**Comment by ozhelezniak-talend** [comment]
Closed because after the rebase PR is a mess.

**Pull request closed without merging**